### PR TITLE
Add score method for linear regression

### LINKED ISSUE #15: MachineLearning_Toolkit: Add Score method for linear regression model
#### Description:

Add method to compute score for the linear regression model. (https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html#sklearn.linear_model.LinearRegression.score)

This method is necessary to evaluate how good the linear regression model perform.

### PR DESCRIPTION
**Pull request by BingWangUS**
### Issues addressed by this PR

Closes #15 

### Test files

https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/MachineLearning_Toolkit/MachineLearning%2315-ScoreMethod?csf=1&web=1&e=JLEdFN

### Changelog

- Add score method under linear regression model

### CODE REVIEW

File: MachineLearning_Engine/Compute/LinearRegression.cs [hunk 1]
```diff
@@ -39,20 +39,22 @@ public static partial class Compute
...
         [Description("Finds the slope and the intercept of a linear function that best fits the given set of bidimensional data.")]
-        [Input("x", "Training data as a list of 2-elements list")]
-        [Input("y", "Target values as a list of 2-elements list")]
-        [Output("model", "The ordinary least squares linear regression with the given data.")]
-        public static LinearRegression LinearRegression(Tensor x, Tensor y)
+        [Input("x", "Training data as a list of 2-elements list.")]
+        [Input("y", "Target values as a list of 2-elements list.")]
+        [MultiOutput(0, "model", "The ordinary least squares linear regression with the given data.")]
+        [MultiOutput(1, "r2", "The coefficient of determination R^2 of the prediction.")]
+        public static Output<LinearRegression, Tensor> LinearRegression(Tensor x, Tensor y)
```
Comment (epignatelli) [inline MachineLearning_Engine/Compute/LinearRegression.cs]: Let's stick to returning only the object we asked for when calling a method with the same name.

This allows keeping things modular.

File: MachineLearning_Engine/Compute/LinearRegression.cs [hunk 2]
```diff
@@ -39,20 +39,22 @@ public static partial class Compute
...
         {
             PyObject model = BH.Engine.MachineLearning.Compute.Invoke("LinearRegression.fit", x, y);
-            return new LinearRegression(model);
+            Tensor score = new Tensor(BH.Engine.MachineLearning.Compute.Invoke("LinearRegression.score", model, x, y));
+            return new Output<LinearRegression, Tensor> { Item1 = new LinearRegression(model), Item2 = score };
```
Comment (epignatelli) [inline MachineLearning_Engine/Compute/LinearRegression.cs]: You can reflecting this using a `Query.Error` method that takes in: the model, the unvisited Xs and the targets (Ys).

File: MachineLearning_Engine/Compute/LinearRegression.py [hunk 3]
```diff
@@ -45,4 +45,12 @@ def infer(model: LinearRegression, x: np.ndarray):
 	# make sure the input is at least bidimensinal
 	if x.ndim == 1:
 		x = x.reshape(-1, 1)
-	return model.predict(x)
\ No newline at end of file
+	return model.predict(x)
+
+def score(model: LinearRegression, x: np.ndarray, y: np.ndarray):
```
Comment (epignatelli) [inline MachineLearning_Engine/Compute/LinearRegression.py]: Can we call this `error`? Or `loss`?

**Review by epignatelli — CHANGES_REQUESTED** [review]
Sorry, It took me a while to review this.

I think the pr is introducing the concept of `error` in the ML toolkit (`score` is nothing else than the MSE of the regression).

Left some comments on the code such that we get it right at the start.

File: MachineLearning_Engine/Query/Error.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        /*************************************/
+
+        [Description("Finds the The coefficient of determination R^2 of the given regression model.")]
+        [Input("model", "The linear regressor model used for inference")]
+        [Input("x", "Training data as a list of 2-elements list.")]
+        [Input("y", "Target values as a list of 2-elements list.")]
+        [Output("r2", "The coefficient of determination R^2 of the prediction.")]
+        public static Tensor Error(LinearRegression model, Tensor x, Tensor y)
+        {
+            return new Tensor(BH.Engine.MachineLearning.Compute.Invoke("LinearRegression.score", model, x, y)); ;
```
Comment (epignatelli) [inline MachineLearning_Engine/Query/Error.cs]: I think the call is to `LinearRegression.error`?
```suggestion
            return new Tensor(BH.Engine.MachineLearning.Compute.Invoke("LinearRegression.error", model, x, y)); ;
```

Comment (BingWangUS) [reply MachineLearning_Engine/Query/Error.cs]: Good catch. My bad.

**Review by epignatelli — APPROVED** [review]
Looks great! Thanks @BingWangUS !
